R/print_to_ggplot.R

```diff
@@ -0,0 +1,158 @@
+#' Print to ggplot
+#'
+#' @param tfrmt tfrmt object that will dictate the structure of the ggplot object
+#' @param .data Data to style in order to make the ggplot object
+#'
+#' @return a stylized ggplot object
+#' @export
+#'
+#' @section Examples:
+#'
+#' ```r
+#'
+#' # Create data
+#' risk<-tibble(time=c(rep(c(0,1000,2000,3000),3)),
+#'             label=c(rep("Obs",4),rep("Lev",4),rep("Lev+5FU",4)),
+#'             value=c(630,372,256,11,620,360,266,8,608,425,328,14),
+#'             param=rep("n",12))
+#'
+#'table<-tfrmt(
+#'  label = label ,
+#'  column = time,
+#'  param = param,
+#'  values = value) %>%
+#'   print_to_ggplot(risk)
+#'
+#'table
+#'
+#' ```
+#' \if{html}{\out{
+#' `r "<img src=\"https://raw.githubusercontent.com/GSK-Biostatistics/tfrmt/master/images/example_print_to_ggplot.png\" style=\"width:50\\%;\">"`
+#' }}
+#'
+#' @importFrom rlang quo_is_missing quo_name
+#' @importFrom dplyr select
+#' @importFrom magrittr %>%
+
+print_to_ggplot <- function(tfrmt, .data){
+  if(!is_tfrmt(tfrmt)){
+    stop("Requires a tfrmt object")
+  }
+
+  if(!is.data.frame(.data)){
+    stop("Requires data")
+  }
+
+  if(is.null(tfrmt$body_plan)==FALSE | is.null(tfrmt$col_plan)==FALSE | is.null(tfrmt$row_grp_plan)==FALSE | is.null(tfrmt$col_style_plan)==FALSE){
+    stop("print_to_ggplot is not currently compatible with Body Plan, Row Group Plan, Column Plan or Column Style Plan. Please remove before continuing.")
```

Comment:
A body plan would mean you can't do any rounding. Is there a reason that isn't allowed? It should be applied in `apply_tfrmt` so I don't think it will cause you problems. similarly row_group_plan shouldn't be an issue as long as the element_row_grp_loc is a indented. I think column plan should also be fine.

But, I like the idea of the check. I think it just needs to be more specific like: 
- element_row_grp_loc must be indented is row_group_plan is present 
- col_plan can't include spanning columns 
- column can only have a single column